### PR TITLE
fix #40: key length comparison in config.c

### LINKED ISSUE #40: awg-quick crashes when i1–i5 parameters are empty (AmneziaWG v2 protocol)
Hi, I'm getting an error when trying to bring up an [AmneziaWG v2 interface on Linux](https://github.com/amnezia-vpn/amneziawg-tools/blob/master/src/wg-quick/linux.bash) using a config exported from Amnezia self-hosted.

The config has `Jc`, `Jmin`, `Jmax`, `S1`, `S2`, `H1`–`H4` set, but `I1`–`I5` come out empty from the export:

```ini
[Interface]
PrivateKey = <key>
Address =
DNS = 1.1.1.1
Jc = 4
Jmin = 40
Jmax = 70
S1 = 0
S2 = 0
H1 = 1
H2 = 2
H3 = 3
H4 = 4
I1 = <b 0x084481800001000300000000077469636b...
I2 = 
I3 =
I4 =
I5 =

[Peer]
PublicKey = <key>
AllowedIPs = 0.0.0.0/0
Endpoint = <host>:<port>
```

Running `awg-quick up awg0` fails. After some digging I found the issue in `awg-quick` — empty `I1`–`I5` values are silently dropped and never passed to `awg setconf`:

```bash
I1|I2|I3|I4|I5) [[ -n $value ]] && WG_CONFIG+="$line"$'\n'; continue ;;
```

So `awg setconf` gets a config without those fields and chokes on it. No warning is printed, which makes it hard to debug.

This seems to affect anyone exporting configs from Amnezia self-hosted since that's where the empty `I1`–`I5` come from.

### PR DESCRIPTION
**Pull request by miuirussia**
We already have checking emptiness of string in parsers, so empty value is valid. Fixes #40

### CODE REVIEW

**Comment by bayes1701** [comment]
Probably it's better to skip lines with empty values in `awg-quick`:

```diff
@@ -49,8 +49,10 @@ parse_options() {
        shopt -s nocasematch
        while read -r line || [[ -n $line ]]; do
                stripped="${line%%\#*}"
+               [[ -z "${stripped//[[:space:]]/}" ]] && continue # skip empty lines
                key="${stripped%%=*}"; key="${key##*([[:space:]])}"; key="${key%%*([[:space:]])}"
                value="${stripped#*=}"; value="${value##*([[:space:]])}"; value="${value%%*([[:space:]])}"
+               [[ $stripped == *"="* && -z "$value" ]] && continue # skip lines with empty values
                [[ $key == "["* ]] && interface_section=0
                [[ $key == "[Interface]" ]] && interface_section=1
                if [[ $interface_section -eq 1 ]]; then
```

The reason is we are setting flags for found keys:
https://github.com/amnezia-vpn/amneziawg-tools/blob/5d6179a6d0842e98dfb349c28cf1bd8e4b9d1079/src/config.c#L580-L584

... and we can set a key unintentionally to an empty string (like `PrivateKey=`)

**Comment by miuirussia** [comment]
no, not better, as some scripts use directly complied binary, not scripts. 

and empty PrivateKey will throw exception `Key is not the correct length or format: `

**Comment by bayes1701** [comment]
I think setting `ctx->device->flags` for empty values may have corner cases

**Comment by miuirussia** [comment]
If you think, so test it and show me cases with error
